Tests/AppcuesKitTests/DecodingExperienceConfig+Testable.swift

```diff
@@ -0,0 +1,174 @@
+//
+//  DecodingExperienceConfig+Testable.swift
+//  AppcuesKitTests
+//
+//  Created by Matt on 2022-12-12.
+//  Copyright © 2022 Appcues. All rights reserved.
+//
+
+import Foundation
+import XCTest
+@testable import AppcuesKit
+
+// Allows initializing a DecodingExperienceConfig from a test with a plain dictionary
+extension DecodingExperienceConfig {
```

Comment:
Just wondering if there is any way to put `DecodingExperienceConfig` behind some Protocol that would allow for more simplified mocking in test, not needing to actually create any `KeyedDecodingContainer` etc. The subscript relies on T being Decodable though, so maybe not.

Reply:
Yeah exactly—the generic subscript complicates things. I also considered letting `DecodingExperienceConfig` store a `[String: Any]` dict, but I really don't like building testing-only codepaths into non-test code.